src/community/oseo/oseo-rest/src/main/java/org/geoserver/opensearch/rest/AbstractOpenSearchController.java

```diff
@@ -217,7 +223,35 @@ protected OgcLinks buildOgcLinksFromFeature(Feature feature, boolean notFoundOnE
         List<OgcLink> links = Collections.emptyList();
         Collection<Property> linkProperties =
                 feature.getProperties(OpenSearchAccess.OGC_LINKS_PROPERTY_NAME);
-        if (linkProperties != null) {
+
+        Map<String, String> unknownFields = new HashMap<>();
+        // taken from OgcLink properties. these properties are known(static) from OgcLink.
+        Set<String> knownFields =
+                new HashSet<>(
+                        Arrays.asList(
+                                "collection_id",
+                                "lid",
+                                "offering",
+                                "method",
+                                "code",
+                                "type",
+                                "href"));
+
+        if (linkProperties != null && linkProperties.size() > 0) {
+            SimpleFeatureImpl simpleFeature =
+                    (SimpleFeatureImpl) ((ArrayList) linkProperties).get(0);
+
+            // iterate over feature to get unknown properties from OgcLink
+            List<AttributeType> types = simpleFeature.getType().getTypes();
+            for (int i = 0; i < types.size(); i++) {
+                String unknownColumnName = types.get(i).getName().getLocalPart();
+                if (!knownFields.contains(unknownColumnName)
+                        && simpleFeature.getAttributes().get(i) != null) {
+                    unknownFields.put(
+                            unknownColumnName, simpleFeature.getAttributes().get(i).toString());
```

Comment:
Hum... string? No, the data type from the database should be preserved:
- When asking for the JSON representation of OGCLink, I expect numbers to be non-quoted, as well as booleans, and dates formatted as ISO dates.
- This code is not null safe, a property can legitimately be null.

Please change your test database to include columns with types int, float, boolean and date. Make sure the types are respected as the data flows across the layers of code, and in the REST API, as well as getting reported properly in templates (JSON templates is type aware too).

Reply:
Changed to the object type. Dates are written in ISO date format.